### ISSUE TITLE
[Question] Is it possible to "include" a file when running checks for unused code?

### ISSUE THREAD

**Issue by dustincatap** [dart-code-checker/dart-code-metrics#694]
**What do you want to discuss?**
I am using [injectable](https://www.pub.dev/packages/injectable). This generates a file that configures the dependency registrations, for my case it is named `service_locator.config.dart`. I also added this to the list of files to be excluded from Dart's analyzer:

```yaml
analyzer:
  exclude:
    - lib/**/*.gen.dart
    - lib/**/*.config.dart
    - lib/**/*.g.dart
    - lib/**/generated/**
```

Here is a sample code of how I use `injectable`:

```dart
@Named(as: 'loginView')
@Injectable(as: Widget)
class LoginView extends StatelessWidget { // dart-code-metrics warns this as unused class
 ...
}
```

And here is a sample content of `service_locator.config.dart`:

```dart
import 'views/login_view.dart' as _i45;

Future<GetIt> $initGetIt(GetIt get, {String? environment, EnvironmentFilter? environmentFilter}) async {
    ...
    gh.factory<Widget>(() => _i45.LoginView(), instanceName: 'loginView');
    ...
}
```

When resolving any registered dependencies, I do not use the class' name, but instead use the instance name:

```dart
// view_locator.dart
Widget getView(String named) => locator<Widget>(instanceName: named);
```

When running `flutter pub run dart_code_metrics:metrics check-unused-code lib --fatal-unused`, here is the output:

```
lib\views\login_view.dart:
    ⚠ unused class LoginView
      at lib\views\login_view.dart:9:1
```

The reason I believe is because `service_locator.config.dart` was excluded from Dart's analyzer. It works fine when I remove it from the excluded files.

Is it possible to include `service_locator.config.dart` when running the above command?

**Comment by incendial** [comment]
@dustincatap hi, interesting, but this looks more like a bug to me, any chance you can create a reproduction example?

**Comment by dustincatap** [comment]
Here is the [sample project](https://github.com/dart-code-checker/dart-code-metrics/files/8064589/sample.zip)

**Comment by incendial** [comment]
@dustincatap thank you so much for the example, I was able to locate the problem and fix it here #697 

Output **before** the fix:
<img width="718" alt="Screenshot 2022-02-15 at 23 18 36" src="https://user-images.githubusercontent.com/21116742/154142309-b008b441-a56b-4035-ae87-1efe5776db10.png">

Output **after** the fix:
<img width="718" alt="Screenshot 2022-02-15 at 23 18 54" src="https://user-images.githubusercontent.com/21116742/154142365-59f46260-be0e-460e-851c-612ddfa9b4c2.png">

Probably will be available in 4.11.0

**Comment by dkrutskikh** [comment]
Available in 4.11.0 release 🚀
